### PR TITLE
don't use ImmutableConverter for nullable structs

### LINKED ISSUE #234: SystemTextJsonSerializer cannot (de)serialize Nullable<...> types
### Exception
```cs
System.InvalidCastException : Unable to cast object of type 'GraphQL.Client.Serializer.SystemTextJson.ImmutableConverter' to type 'System.Text.Json.Serialization.JsonConverter`1[System.Nullable`1[System.Int32]]'.
   at System.Text.Json.JsonPropertyInfoCommon`4.set_ConverterBase(JsonConverter value)
   at System.Text.Json.JsonPropertyInfo.Initialize(Type parentClassType, Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type elementType, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoCommon`4.Initialize(Type parentClassType, Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type elementType, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.CreateProperty(Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type parentClassType, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.AddProperty(Type propertyType, PropertyInfo propertyInfo, Type classType, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClass(Type classType)
   at System.Text.Json.JsonSerializer.GetRuntimeClassInfo(Object value, JsonClassInfo& jsonClassInfo, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.HandleDictionary(JsonClassInfo elementClassInfo, JsonSerializerOptions options, Utf8JsonWriter writer, WriteStack& state)
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at GraphQL.Client.Serializer.SystemTextJson.SystemTextJsonSerializer.SerializeToString(GraphQLRequest request)
   at GraphQL.Client.Http.GraphQLHttpRequest.ToHttpRequestMessage(GraphQLHttpClientOptions options, IGraphQLJsonSerializer serializer)
   at GraphQL.Client.Http.GraphQLHttpClient.SendHttpRequestAsync[TResponse](GraphQLRequest request, CancellationToken cancellationToken)
   at GraphQL.Client.Http.GraphQLHttpClient.SendQueryAsync[TResponse](GraphQLRequest request, CancellationToken cancellationToken)
...
```

### Reproducing the bug
```cs
        public class WithNullable
        {
            public int? NullableInt { get; set; }
        }

        [Test]
        public void ReproductionImmutableConverterBug()
        {
            var systemTextJsonSerializer = new SystemTextJsonSerializer();
            var serializedString = systemTextJsonSerializer.SerializeToString(new GraphQLRequest
            {
                Query = "{}",
                Variables = new WithNullable
                {
                    NullableInt = 2
                }
            });
        }
```

### PR DESCRIPTION
**Pull request by wirmar**
fixes #234

### CODE REVIEW

File: src/GraphQL.Client.Serializer.SystemTextJson/ImmutableConverter.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public override bool CanConvert(Type typeToConvert)
                 return false;
 
             var nullableUnderlyingType = Nullable.GetUnderlyingType(typeToConvert);
-            if (nullableUnderlyingType != null && nullableUnderlyingType.IsPrimitive)
+            if (nullableUnderlyingType != null && nullableUnderlyingType.IsValueType)
```
Comment (sungam3r) [inline src/GraphQL.Client.Serializer.SystemTextJson/ImmutableConverter.cs:22]: 😄

Comment (wirmar) [reply src/GraphQL.Client.Serializer.SystemTextJson/ImmutableConverter.cs:22]: What exactly do you mean by this review? 😄

**Comment by rose-a** [comment]
Thanks for fixing this...

File: src/GraphQL.Client.Serializer.SystemTextJson/ImmutableConverter.cs (hunk 1 above)
Comment (sungam3r) [reply src/GraphQL.Client.Serializer.SystemTextJson/ImmutableConverter.cs:22]: The devil is in the little things.

**Pull request merged**